exchange/bitswap/decision/strategy_prq.go

```diff
@@ -0,0 +1,225 @@
+// Strategy PRQ
+// ============
+
+package decision
+
+import (
+	"sync"
+	"time"
+
+	wantlist "github.com/ipfs/go-ipfs/exchange/bitswap/wantlist"
+	pq "github.com/ipfs/go-ipfs/thirdparty/pq"
+
+	peer "gx/ipfs/QmWNY7dV54ZDYmTA1ykVdwNCqC11mpU4zSUp6XDpLTH9eG/go-libp2p-peer"
+	cid "gx/ipfs/QmeSrf6pzut73u6zLQkRFQ3ygt3k6XFT2kjdYP8Tnkwwyg/go-cid"
+)
+
+// Type and Constructor
+// --------------------
+
+// verify interface implementation
+var _ peerRequestQueue = &strategy_prq{}
+
+type strategy_prq struct {
+	lock     sync.Mutex
+	pQueue   pq.PQ
+	taskMap  map[string]*peerRequestTask
+	partners map[peer.ID]*activePartner
+	rrq      *RRQueue
+}
+
+func newStrategyPRQ(strategy Strategy) *strategy_prq {
+	return &strategy_prq{
+		taskMap:  make(map[string]*peerRequestTask),
+		partners: make(map[peer.ID]*activePartner),
+		pQueue:   pq.New(partnerCompare),
```

Comment:
`partnerQueue`?

Reply:
Yeah, I named it `pQueue` to maintain symmetry with the old PRQ. I can do another PR after and change the name in both.